velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -1383,6 +1385,73 @@ TpchPlan TpchQueryBuilder::getQ16Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ17Plan() const {
+  std::vector<std::string> lineitemColumns = {
+      "l_partkey", "l_extendedprice", "l_quantity"};
+  std::vector<std::string> partColumns = {
+      "p_partkey", "p_brand", "p_container"};
+
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  auto partSelectedRowType = getRowType(kPart, partColumns);
+  const auto& partFileColumns = getFileColumnNames(kPart);
+
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  core::PlanNodeId lineitemScanNodeId;
+  core::PlanNodeId lineitemScanNodeIdSubQuery;
+  core::PlanNodeId partScanNodeId;
+
+  auto part = PlanBuilder(planNodeIdGenerator)
+                  .tableScan(
+                      kPart,
+                      partSelectedRowType,
+                      partFileColumns,
+                      {"p_brand = 'Brand#23'", "p_container = 'MED BOX'"})
+                  .capturePlanNodeId(partScanNodeId)
+                  .planNode();
+
+  auto lineitemJoinPart =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kLineitem, lineitemSelectedRowType, lineitemFileColumns)
+          .capturePlanNodeId(lineitemScanNodeId)
+          .hashJoin(
+              {"l_partkey"},
+              {"p_partkey"},
+              part,
+              "",
+              {"l_quantity", "p_partkey", "l_extendedprice"})
+          .planNode();
+
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kLineitem, lineitemSelectedRowType, lineitemFileColumns)
+          .capturePlanNodeId(lineitemScanNodeIdSubQuery)
+          .partialAggregation({"l_partkey"}, {"avg(l_quantity) as avg_"})
+          .localPartition({})
+          .finalAggregation()
+          .hashJoin(
+              {"l_partkey"},
+              {"p_partkey"},
+              lineitemJoinPart,
+              "",
+              {"l_extendedprice", "l_quantity", "avg_"},
+              core::JoinType::kRight)
```

Comment:
Why right join? Rows will null on the left side won't pass the filter that follows? Can this be an inner join with a filter evaluated as part of the join itself?

Reply:
Removed the right join with the new plan as suggested, thanks!